sqlmesh/dbt/basemodel.py

```diff
@@ -296,22 +304,52 @@ def __init__(self, context: DbtContext, config: BMC):
     @property
     def enriched_config(self) -> BMC:
         if self._rendered_sql is None:
+            self._enriched_config = self._update_with_sql_config(self._enriched_config)
             self.render()
             self._enriched_config._dependencies = self._enriched_config._dependencies.union(
                 self._captured_dependencies
             )
         return self._enriched_config
 
+    @property
+    def parsed_sql(self) -> nodes.Template:
+        if self._parsed_sql is None:
+            self._parsed_sql = JINJA_ENVIRONMENT.parse(self.config.all_sql)
+        return self._parsed_sql
+
     def render(self) -> str:
         if self._rendered_sql is None:
             registry = self.context.jinja_macros
             self._rendered_sql = (
                 registry.build_environment(**self._jinja_globals)
-                .from_string(self.config.all_sql)
+                .from_string(self.parsed_sql)
                 .render()
             )
         return self._rendered_sql
 
+    def _update_with_sql_config(self, config: BMC) -> BMC:
+        def _extract_value(node: t.Any) -> t.Any:
+            if not isinstance(node, nodes.Node):
+                return node
+            if isinstance(node, nodes.Const):
+                return _extract_value(node.value)
+            if isinstance(node, nodes.List):
+                return [_extract_value(val) for val in node.items]
+            if isinstance(node, nodes.Dict):
+                return {_extract_value(pair.key): _extract_value(pair.value) for pair in node.items}
+            if isinstance(node, nodes.Tuple):
+                return tuple(_extract_value(val) for val in node.items)
+            raise SQLMeshError(f"Unable to extract node type '{type(node)}'.")
```

Comment:
Aren't we suppose to evaluate an expression if it's not a string literal? Eg. how does this work if there is a nested function call, eg:
```
config(
  attr=fun(...)
)
```

Reply:
Good catch